Parser divergence: multiline strings in JSX attributes

When strings are directly used in the right side of a JSX attribute, they may be multiline. [These strings have a different production rule](https://facebook.github.io/jsx/#prod-JSXAttributeValue) in the JSX spec.

```tsx
export let x = <div className="align-top
  align-right" />;
```

**Expected**: No errors
**Actual Behavior**:

```
bar.tsx:1:41 - error TS1002: Unterminated string literal.

1 export let x = <div className="align-top
                                          

bar.tsx:2:14 - error TS1003: Identifier expected.

2   align-right" />
               ~~~~

bar.tsx:2:18 - error TS1002: Unterminated string literal.

2   align-right" />
                   


Found 3 errors in the same file, starting at: bar.tsx:1
```

--- a/internal/scanner/scanner.go
+++ b/internal/scanner/scanner.go
@@ -1056,7 +1056,7 @@ func (s *Scanner) scanString(jsxAttributeString bool) string {
 			start = s.pos
 			continue
 		}
-		if ch == '\n' || ch == '\r' && !jsxAttributeString {
+		if (ch == '\n' || ch == '\r') && !jsxAttributeString {
 			sb.WriteString(s.text[start:s.pos])
 			s.tokenFlags |= ast.TokenFlagsUnterminated
 			s.error(diagnostics.Unterminated_string_literal)

